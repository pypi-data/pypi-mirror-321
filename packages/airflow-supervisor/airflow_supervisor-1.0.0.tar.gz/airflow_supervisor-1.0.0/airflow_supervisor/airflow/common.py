from typing import Literal

from airflow.exceptions import AirflowSkipException

SupervisorTaskStep = Literal[
    "configure-supervisor",
    "start-supervisor",
    "start-programs",
    "stop-programs",
    "check-programs",
    "restart-programs",
    "stop-supervisor",
    "unconfigure-supervisor",
    "force-kill",
]


def skip_():
    raise AirflowSkipException


__all__ = (
    "SupervisorTaskStep",
    "skip_",
)
