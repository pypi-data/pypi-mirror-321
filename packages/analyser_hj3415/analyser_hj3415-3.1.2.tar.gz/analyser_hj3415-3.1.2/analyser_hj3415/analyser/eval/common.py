import math
from typing import Tuple

from db_hj3415 import myredis
from utils_hj3415.tools import nan_to_zero
from utils_hj3415 import setup_logger

mylogger = setup_logger(__name__,'WARNING')


class Tools:
    @staticmethod
    def sigmoid_score(deviation, a=1.0, b=2.0):
        """
        Calculates a normalized score using a sigmoid function based on the provided deviation value.

        This method applies the sigmoid function to a logarithmically transformed deviation value
        to map it to a range between 0 and 100. The shape of the sigmoid curve can be adjusted
        with parameters `a` and `b`.

        Parameters:
            deviation (float): The deviation value to be transformed. Must be a non-negative value.
            a (float): The steepness of the sigmoid curve. Default is 1.0.
            b (float): The x-offset for the sigmoid curve. Default is 2.0.

        Returns:
            float: A score between 0 and 100 derived from the provided deviation value.
        """
        # 예: x = log10(deviation + 1)
        x = math.log10(deviation + 1)
        s = 1 / (1 + math.exp(-a * (x - b)))  # 0~1 범위
        return s * 100  # 0~100 범위

    @staticmethod
    def log_score(deviation):
        """
            Compute and return the logarithmic score scaled by a constant factor.

            This method takes a numerical deviation value, adds one to it, computes its
            base-10 logarithm, and then multiplies the result by a constant factor of 33
            to scale the resulting logarithmic score.

            Parameters:
                deviation (float): The numerical deviation value to calculate the
                logarithmic score for. Should be a non-negative number.

            Returns:
                float: The scaled logarithmic score computed based on the input deviation.
        """
        return math.log10(deviation + 1) * 33

    @staticmethod
    def cal_deviation(v1: float, v2: float) -> float:
        """
            Calculates the percentage deviation between two values.

            This method computes the percentage deviation of the second value
            from the first value based on the formula:
            deviation = abs((v1 - v2) / v1) * 100. In the event the first value is
            zero (division by zero), the function will return NaN to signify
            an invalid computation.

            Parameters:
            v1 (float): The reference value. It represents the base for the relative
                        deviation calculation.
            v2 (float): The value to compare against the reference.

            Returns:
            float: The computed percentage deviation. Returns NaN if the reference
                   value (v1) is zero.
        """
        try:
            deviation = abs((v1 - v2) / v1) * 100
        except ZeroDivisionError:
            deviation = math.nan
        return deviation

    @staticmethod
    def date_set(*args) -> list:
        """
            인자로 받은 값의 비유효한 내용 제거(None,nan)하고 중복된 항목 제거하고 리스트로 반환한다.

            여기서 set의 의미는 집합을 뜻함

            Filters and returns a list of unique non-null, non-empty values from
            the provided arguments.

            This static method processes the input arguments to retain only unique
            values that are not empty strings, NaN values, or None. The result is
            returned as a list.

            Args:
                *args: Arbitrary positional arguments to be filtered.

            Returns:
                list: A list of unique values after filtering out invalid entries.
        """
        return [i for i in {*args} if i != "" and i is not math.nan and i is not None]

    @staticmethod
    def calc당기순이익(c103: myredis.C103, refresh: bool) -> Tuple[str, float]:
        """
            지배지분 당기순이익 계산

            일반적인 경우로는 직전 지배주주지분 당기순이익을 찾아서 반환한다.

            금융기관의 경우는 지배당기순이익이 없기 때문에 계산을 통해서 간접적으로 구한다.

            Calculates "지배당기순이익" (Controlling Comprehensive Income) based on the given
            financial data. The method retrieves or computes the value utilizing methods from
            the `myredis.C103` class. It handles missing or 'Not-a-Number' conditions by
            manually calculating from quarterly and annual financial figures. Logs the process
            at various stages for debugging and auditing.

            Args:
                c103 (myredis.C103): An instance containing financial data and utilities to
                    access specific data points for the targeted calculation.
                refresh (bool): A flag to determine whether or not to refresh the data
                    while accessing or computing financial values.

            Returns:
                Tuple[str, float]: A tuple where the first item is the most relevant date for
                    the calculated or retrieved value, and the second item is the calculated
                    or retrieved "지배당기순이익" (Controlling Comprehensive Income).
        """
        name = myredis.Corps(c103.code, 'c101').get_name(refresh=refresh)

        mylogger.info(f'{c103.code} / {name} Tools : 당기순이익 계산.. refresh : {refresh}')
        c103.page = 'c103재무상태표q'

        d1, 지배당기순이익 = c103.latest_value_pop2('*(지배)당기순이익', refresh)
        mylogger.debug(f"*(지배)당기순이익: {지배당기순이익}")

        if math.isnan(지배당기순이익):
            mylogger.warning(f"{c103.code} / {name} - (지배)당기순이익이 없는 종목. 수동으로 계산합니다.")
            c103.page = 'c103손익계산서q'
            d2, 최근4분기당기순이익 = c103.sum_recent_4q('당기순이익', refresh)
            mylogger.debug(f"{c103.code} / {name} - 최근4분기당기순이익 : {최근4분기당기순이익}")
            c103.page = 'c103재무상태표y'
            d3, 비지배당기순이익 = c103.latest_value_pop2('*(비지배)당기순이익', refresh)
            mylogger.debug(f"{c103.code} / {name} - 비지배당기순이익y : {비지배당기순이익}")
            # 가변리스트 언패킹으로 하나의 날짜만 사용하고 나머지는 버린다.
            # 여기서 *_는 “나머지 값을 다 무시하겠다”는 의미
            mylogger.debug(f"d2:{d2}, d3: {d3}")
            try:
                date, *_ = Tools.date_set(d2, d3)
            except ValueError:
                # 날짜 데이터가 없는경우
                date = ''
            계산된지배당기순이익 = round(최근4분기당기순이익 - nan_to_zero(비지배당기순이익), 1)
            mylogger.debug(f"{c103.code} / {name} - 계산된 지배당기순이익 : {계산된지배당기순이익}")
            return date, 계산된지배당기순이익
        else:
            return d1, 지배당기순이익

    @staticmethod
    def calc유동자산(c103: myredis.C103, refresh: bool) -> Tuple[str, float]:
        """
            유효한 유동자산 계산

            일반적인 경우로 유동자산을 찾아서 반환한다.

            금융기관의 경우는 간접적으로 계산한다.

            Calculates the current assets for a given company code.

            For a specified company, the function calculates the recent 4-quarter
            sum of current assets if available. If the data is not available or
            contains invalid values, it attempts to calculate the current assets
            manually using financial asset data such as cash equivalents, trading
            securities, available-for-sale securities, and held-to-maturity securities.

            Logs relevant information and warnings during the calculation process,
            including any cases where data is unavailable or a manual calculation
            is required.

            Parameters:
                c103 (myredis.C103): The instance representing financial data of a
                    specific company. This includes methods to extract and calculate
                    various data points.
                refresh (bool): Indicator flag to determine whether to refresh the
                    underlying data before performing calculations.

            Returns:
                Tuple[str, float]: A tuple containing the date associated with the
                    financial data and the calculated or retrieved value of current
                    assets. If dates are not available, the date field may be empty.
        """

        name = myredis.Corps(c103.code, 'c101').get_name(refresh=refresh)

        mylogger.info(f'{c103.code} / {name} Tools : 유동자산계산... refresh : {refresh}')
        c103.page = 'c103재무상태표q'

        d, 유동자산 = c103.sum_recent_4q('유동자산', refresh)
        if math.isnan(유동자산):
            mylogger.warning(f"{c103.code} / {name} - 유동자산이 없는 종목. 수동으로 계산합니다(금융관련업종일 가능성있음).")
            d1, v1 = c103.latest_value_pop2('현금및예치금', refresh)
            d2, v2 = c103.latest_value_pop2('단기매매금융자산', refresh)
            d3, v3 = c103.latest_value_pop2('매도가능금융자산', refresh)
            d4, v4 = c103.latest_value_pop2('만기보유금융자산', refresh)
            mylogger.debug(f'{c103.code} / {name} 현금및예치금 : {d1}, {v1}')
            mylogger.debug(f'{c103.code} / {name} 단기매매금융자산 : {d2}, {v2}')
            mylogger.debug(f'{c103.code} / {name} 매도가능금융자산 : {d3}, {v3}')
            mylogger.debug(f'{c103.code} / {name} 만기보유금융자산 : {d4}, {v4}')

            try:
                date, *_ = Tools.date_set(d1, d2, d3, d4)
            except ValueError:
                # 날짜 데이터가 없는경우
                date = ''
            계산된유동자산value = round(
                nan_to_zero(v1) + nan_to_zero(v2) + nan_to_zero(v3) + nan_to_zero(v4), 1)

            mylogger.info(f"{c103.code} / {name} - 계산된 유동자산 : {계산된유동자산value}")
            return date, 계산된유동자산value
        else:
            return d, 유동자산

    @staticmethod
    def calc유동부채(c103: myredis.C103, refresh: bool) -> Tuple[str, float]:
        """
            유효한 유동부채 계산

            일반적인 경우로 유동부채를 찾아서 반환한다.

            금융기관의 경우는 간접적으로 계산한다.

            Calculate '유동부채' (Current Liabilities) based on financial data of a specific entity.

            This static method computes the recent '유동부채' value either from the sum of recent four
            quarters using predefined keys or calculates manually if no valid data is available.
            It includes logging for intermediate steps and supports handling missing values by logging
            warnings and attempting a composed manual computation using alternative financial terms.

            Args:
                c103 (myredis.C103): The object containing financial data and operations for obtaining the required data.
                refresh (bool): A flag to indicate whether to fetch the latest data forcibly.

            Returns:
                Tuple[str, float]: A tuple containing the `date` of financial data and the computed '유동부채' value.
        """

        name = myredis.Corps(c103.code, 'c101').get_name(refresh=refresh)

        mylogger.info(f'{c103.code} / {name} Tools : 유동부채계산... refresh : {refresh}')
        c103.page = 'c103재무상태표q'

        d, 유동부채 = c103.sum_recent_4q('유동부채', refresh)
        if math.isnan(유동부채):
            mylogger.warning(f"{c103.code} / {name} - 유동부채가 없는 종목. 수동으로 계산합니다.")
            d1, v1 = c103.latest_value_pop2('당기손익인식(지정)금융부채', refresh)
            d2, v2 = c103.latest_value_pop2('당기손익-공정가치측정금융부채', refresh)
            d3, v3 = c103.latest_value_pop2('매도파생결합증권', refresh)
            d4, v4 = c103.latest_value_pop2('단기매매금융부채', refresh)
            mylogger.debug(f'{c103.code} / {name} 당기손익인식(지정)금융부채 : {d1}, {v1}')
            mylogger.debug(f'{c103.code} / {name} 당기손익-공정가치측정금융부채 : {d2}, {v2}')
            mylogger.debug(f'{c103.code} / {name} 매도파생결합증권 : {d3}, {v3}')
            mylogger.debug(f'{c103.code} / {name} 단기매매금융부채 : {d4}, {v4}')

            try:
                date, *_ = Tools.date_set(d1, d2, d3, d4)
            except ValueError:
                # 날짜 데이터가 없는경우
                date = ''
            계산된유동부채value = round(
                nan_to_zero(v1) + nan_to_zero(v2) + nan_to_zero(v3) + nan_to_zero(v4), 1)

            mylogger.info(f"{c103.code} / {name} - 계산된 유동부채 : {계산된유동부채value}")
            return date, 계산된유동부채value
        else:
            return d, 유동부채


"""
- 각분기의 합이 연이 아닌 타이틀(즉 sum_4q를 사용하면 안됨)
'*(지배)당기순이익'
'*(비지배)당기순이익'
'장기차입금'
'현금및예치금'
'매도가능금융자산'
'매도파생결합증권'
'만기보유금융자산'
'당기손익-공정가치측정금융부채'
'당기손익인식(지정)금융부채'
'단기매매금융자산'
'단기매매금융부채'
'예수부채'
'차입부채'
'기타부채'
'보험계약부채(책임준비금)'
'*CAPEX'
'ROE'
"""

"""
- sum_4q를 사용해도 되는 타이틀
'자산총계'
'당기순이익'
'유동자산'
'유동부채'
'비유동부채'

'영업활동으로인한현금흐름'
'재무활동으로인한현금흐름'
'ROIC'
"""
