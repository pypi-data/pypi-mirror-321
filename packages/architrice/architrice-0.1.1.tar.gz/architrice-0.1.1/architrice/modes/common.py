import logging
import os

from .. import database
from .. import sources
from .. import targets
from .. import utils

from . import cli


def get_source(name, picker=False):
    if name is not None:
        if not isinstance(name, str):
            return name

        try:
            return sources.get(name, True)
        except ValueError as e:
            logging.error(str(e))
    if picker:
        return source_picker()
    return None


def source_picker():
    return cli.get_choice(
        [s.NAME for s in sources.sourcelist],
        "Download from which supported decklist website?",
        sources.sourcelist,
    )()


def get_target(name, picker=False):
    if name is not None:
        if not isinstance(name, str):
            return name

        try:
            return targets.get(name)
        except ValueError as e:
            logging.error(str(e))
    if picker:
        return target_picker()
    return None


def target_picker():
    return cli.get_choice(
        [t.NAME for t in targets.targetlist],
        "For which supported MtG client?",
        targets.targetlist,
    )()


def get_profile(cache, interactive, prompt="Choose a profile"):
    if not cache.profiles:
        logging.error("No profiles, unable to select one.")
        return None

    if len(cache.profiles) == 1:
        logging.info("Defaulted to only profile which matches criteria.")
        return cache.profiles[0]
    elif interactive:
        return cli.get_choice(
            [str(p) for p in cache.profiles], prompt, cache.profiles
        )
    else:
        logging.error("Multiple profiles match criteria.")
        return None


def get_verified_user(source, user, interactive=False):
    if not user:
        if interactive:
            user = cli.get_string(source.name + " username")
        else:
            return None

    if not (
        database.select_one("users", source=source.short, name=user)
        or source.verify_user(user)
    ):
        if interactive:
            print("Couldn't find any public decks for this user. Try again.")
            return get_verified_user(source, None, True)
        else:
            return None
    return user


def get_output_path(cache, interactive, target, path):
    if path is not None:
        if utils.check_dir(path):
            return path
        else:
            logging.error(
                f"A file exists at {path} so it can't be used as an output "
                "directory."
            )
            if not interactive:
                return None
            path = None

    existing_output_dirs = cache.get_all_output_dirs()
    if existing_output_dirs and cli.get_decision(
        "Use existing output directory?"
    ):
        if len(existing_output_dirs) == 1:
            path = existing_output_dirs[0].path
            logging.info(f"Only one existing directory, defaulting to {path}.")
        else:
            path = cli.get_choice(
                [d.path for d in existing_output_dirs],
                "Which existing directory should be used for these decks?",
            )
    else:
        path = target.suggest_directory()
        if not (
            (os.path.isdir(path))
            and cli.get_decision(
                f"Found {target.name} deck directory at {path}."
                " Output decklists here?"
            )
        ):
            return get_output_path(
                cache, interactive, target, cli.get_path("Output directory")
            )
    return path


def add_output(
    cache, interactive, profile, target=None, path=None, include_maybe=None
):
    if profile is None:
        profile = get_profile(
            cache, interactive, "Add an output to which profile?"
        )
        if not profile:
            logging.error("No profile specified. Unable to add output.")
            return

    target = get_target(target, interactive)
    if not target:
        logging.error("No target specified. Unable to add output.")
        return

    path = get_output_path(cache, interactive, target, path)
    if not path:
        logging.error("No path specified. Unable to add output.")
        return

    if include_maybe is None:
        include_maybe = cli.get_decision(
            "Include maybeboards in the decks downloaded?"
        )

    cache.build_output(profile, target, path, include_maybe)
