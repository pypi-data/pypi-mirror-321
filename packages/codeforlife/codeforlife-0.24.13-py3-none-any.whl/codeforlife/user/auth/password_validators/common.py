# pylint: skip-file
# TODO: https://docs.djangoproject.com/en/3.2/topics/auth/passwords/#writing-your-own-validator
class CommonPasswordValidator:
    pass
