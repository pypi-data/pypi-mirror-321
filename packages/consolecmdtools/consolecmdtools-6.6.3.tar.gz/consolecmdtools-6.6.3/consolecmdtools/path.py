import os


class Path(str):
    """A class that represents a path. It is a subclass of str, so it can be used as a string.

    Attributes:
        path (str): The path as a string.
        abs (str): The absolute path as a string.
        basename (str): The basename of the path.
        ext (str): The extension of the path.
        stem (str): The stem of the path.
        parent (Path): The parent directory of the path. Also a Path object.
        exists (bool): True if the path exists, False otherwise.
        is_dir (bool): True if the path is a directory, False otherwise.
        is_file (bool): True if the path is a file, False otherwise.

    Examples:
        filepath: './filename.txt'
        abs: '/path/to/filename.txt'
        basename: 'filename.txt'
        ext: 'txt'
        stem: 'filename'
        parent: '/path/to'
        parent.basename: 'to'
        parent.ext: ''
        parent.stem: 'to'
        parent.parent: '/path'
        +---------------------------+
        |           Path            |
        +----------+----------------+
        |  Parent  |    Basename    |
        |          +----------+-----+
        |          |   Stem   | Ext |
        |          |          |     |
        " /path/to / filename . txt "
        |          |          |     |
        +----------+----------+-----+
    """

    def __init__(self, path: str):
        super().__init__()
        self.path = path

    def __str__(self) -> str:
        return self.abs

    def __repr__(self) -> str:
        return f"Path({self.path!r})"

    @property
    def abs(self) -> str:
        path = self.path
        path = os.path.expanduser(path)
        path = os.path.expandvars(path)
        path = os.path.abspath(path)
        return path

    @property
    def parent(self) -> 'Path':
        return Path(os.path.dirname(self.abs))

    @property
    def basename(self) -> str:
        """Returns the basename of the path."""
        return os.path.basename(self.abs)

    @property
    def ext(self) -> str:
        """Returns the extension of the path without the dot."""
        return os.path.splitext(self.basename)[1].lstrip(".")

    @property
    def stem(self) -> str:
        """Returns the basename of the path but without the extension and the dot."""
        return os.path.splitext(self.basename)[0]

    @property
    def exists(self) -> bool:
        """Returns whether the path exists."""
        return os.path.exists(self.abs)

    @property
    def is_dir(self) -> bool:
        """Returns whether the path is a directory."""
        return os.path.isdir(self.abs)

    @property
    def is_file(self) -> bool:
        """Returns whether the path is a file."""
        return os.path.isfile(self.abs)
