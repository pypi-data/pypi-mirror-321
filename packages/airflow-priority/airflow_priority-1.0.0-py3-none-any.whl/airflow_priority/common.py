import os
from pathlib import Path
from types import MappingProxyType
from typing import Literal, Optional, Tuple

from airflow.models.dagrun import DagRun

__all__ = (
    "has_priority_tag",
    "get_config_option",
    "DagStatus",
    "PriorityTags",
    "AirflowPriorityConfigurationOptionNotFound",
)


DagStatus = Literal["running", "success", "failed"]

PriorityTags = MappingProxyType(
    {
        "P1": 1,
        "P2": 2,
        "P3": 3,
        "P4": 4,
        "P5": 5,
    }
)


class AirflowPriorityConfigurationOptionNotFound(RuntimeError): ...


def get_config_option(section, key, required=True, default=None):
    try:
        from airflow_config import ConfigNotFoundError, Configuration

        try:
            config = Configuration.load("config", "config", basepath=str(Path(os.environ.get("AIRFLOW_HOME", "")) / "dags"), _offset=4)
            ret = getattr(getattr(config.extensions.get("priority", None), section, None), key, None)
            if ret is not None:
                return ret
        except ConfigNotFoundError:
            # SKIP
            pass
    except ImportError:
        # SKIP
        pass

    try:
        import airflow.configuration

        config_option = airflow.configuration.conf.get(f"priority.{section}", key, default)
        if not config_option and required:
            raise AirflowPriorityConfigurationOptionNotFound(f"{section}.{key}")
        return config_option
    except Exception:
        raise AirflowPriorityConfigurationOptionNotFound(f"{section}.{key}")


def has_priority_tag(dag_run: DagRun) -> Optional[Tuple[str, int]]:
    dag = dag_run.dag
    dag_id = dag_run.dag_id
    tags = dag.tags
    if tags:
        for tag in tags:
            priority = PriorityTags.get(tag.upper())
            if priority:
                return dag_id, priority
    return dag_id, None
