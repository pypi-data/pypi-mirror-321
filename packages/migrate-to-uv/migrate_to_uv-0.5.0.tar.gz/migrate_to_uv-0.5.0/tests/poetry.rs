use crate::common::{apply_lock_filters, cli, LockedPackage, UvLock};
use insta_cmd::assert_cmd_snapshot;
use std::path::Path;
use std::{env, fs};
use tempfile::tempdir;

mod common;

const FIXTURES_PATH: &str = "tests/fixtures/poetry";

#[test]
fn test_complete_workflow() {
    let fixture_path = Path::new(FIXTURES_PATH).join("with_lock_file");

    let tmp_dir = tempdir().unwrap();
    let project_path = tmp_dir.path();

    for file in ["poetry.lock", "poetry.toml", "pyproject.toml"] {
        fs::copy(fixture_path.join(file), project_path.join(file)).unwrap();
    }

    apply_lock_filters!();
    assert_cmd_snapshot!(cli().arg(project_path), @r###"
    success: true
    exit_code: 0
    ----- stdout -----

    ----- stderr -----
    Locking dependencies with "uv lock"...
    Using [PYTHON_INTERPRETER]
    Resolved [PACKAGES] packages in [TIME]
    Locking dependencies with "uv lock" again to remove constraints...
    Using [PYTHON_INTERPRETER]
    Resolved [PACKAGES] packages in [TIME]
    Successfully migrated project from Poetry to uv!
    "###);

    insta::assert_snapshot!(fs::read_to_string(project_path.join("pyproject.toml")).unwrap());

    let uv_lock = toml::from_str::<UvLock>(
        fs::read_to_string(project_path.join("uv.lock"))
            .unwrap()
            .as_str(),
    )
    .unwrap();

    // Assert that locked versions in `uv.lock` match what was in `poetry.lock`.
    let uv_lock_packages = uv_lock.package.unwrap();
    let expected_locked_packages = Vec::from([
        LockedPackage {
            name: "arrow".to_string(),
            version: "1.2.3".to_string(),
        },
        LockedPackage {
            name: "factory-boy".to_string(),
            version: "3.2.1".to_string(),
        },
        LockedPackage {
            name: "faker".to_string(),
            version: "33.1.0".to_string(),
        },
        LockedPackage {
            name: "foo".to_string(),
            version: "0.0.1".to_string(),
        },
        LockedPackage {
            name: "mypy".to_string(),
            version: "1.13.0".to_string(),
        },
        LockedPackage {
            name: "mypy-extensions".to_string(),
            version: "1.0.0".to_string(),
        },
        LockedPackage {
            name: "python-dateutil".to_string(),
            version: "2.7.0".to_string(),
        },
        LockedPackage {
            name: "six".to_string(),
            version: "1.15.0".to_string(),
        },
        LockedPackage {
            name: "typing-extensions".to_string(),
            version: "4.6.0".to_string(),
        },
    ]);
    for package in expected_locked_packages {
        assert!(uv_lock_packages.contains(&package));
    }

    // Assert that previous package manager files are correctly removed.
    assert!(!project_path.join("poetry.lock").exists());
    assert!(!project_path.join("poetry.toml").exists());
}

#[test]
fn test_ignore_locked_versions() {
    let fixture_path = Path::new(FIXTURES_PATH).join("with_lock_file");

    let tmp_dir = tempdir().unwrap();
    let project_path = tmp_dir.path();

    for file in ["poetry.lock", "poetry.toml", "pyproject.toml"] {
        fs::copy(fixture_path.join(file), project_path.join(file)).unwrap();
    }

    apply_lock_filters!();
    assert_cmd_snapshot!(cli().arg(project_path).arg("--ignore-locked-versions"), @r###"
    success: true
    exit_code: 0
    ----- stdout -----

    ----- stderr -----
    Locking dependencies with "uv lock"...
    Using [PYTHON_INTERPRETER]
    Resolved [PACKAGES] packages in [TIME]
    Successfully migrated project from Poetry to uv!
    "###);

    insta::assert_snapshot!(fs::read_to_string(project_path.join("pyproject.toml")).unwrap());

    let uv_lock = toml::from_str::<UvLock>(
        fs::read_to_string(project_path.join("uv.lock"))
            .unwrap()
            .as_str(),
    )
    .unwrap();

    let mut arrow: Option<LockedPackage> = None;
    let mut typing_extensions: Option<LockedPackage> = None;
    for package in uv_lock.package.unwrap() {
        if package.name == "arrow" {
            arrow = Some(package);
        } else if package.name == "typing-extensions" {
            typing_extensions = Some(package);
        }
    }

    // Assert that locked versions are different that what was in `poetry.lock`.
    assert_ne!(arrow.unwrap().version, "1.2.3");
    assert_ne!(typing_extensions.unwrap().version, "4.6.0");

    // Assert that previous package manager files are correctly removed.
    assert!(!project_path.join("poetry.lock").exists());
    assert!(!project_path.join("poetry.toml").exists());
}

#[test]
fn test_keep_current_data() {
    let fixture_path = Path::new(FIXTURES_PATH).join("with_lock_file");

    let tmp_dir = tempdir().unwrap();
    let project_path = tmp_dir.path();

    for file in ["poetry.lock", "poetry.toml", "pyproject.toml"] {
        fs::copy(fixture_path.join(file), project_path.join(file)).unwrap();
    }

    apply_lock_filters!();
    assert_cmd_snapshot!(cli().arg(project_path).arg("--keep-current-data"), @r###"
    success: true
    exit_code: 0
    ----- stdout -----

    ----- stderr -----
    Locking dependencies with "uv lock"...
    Using [PYTHON_INTERPRETER]
    Resolved [PACKAGES] packages in [TIME]
    Locking dependencies with "uv lock" again to remove constraints...
    Using [PYTHON_INTERPRETER]
    Resolved [PACKAGES] packages in [TIME]
    Successfully migrated project from Poetry to uv!
    "###);

    insta::assert_snapshot!(fs::read_to_string(project_path.join("pyproject.toml")).unwrap());

    // Assert that previous package manager files have not been removed.
    assert!(project_path.join("poetry.lock").exists());
    assert!(project_path.join("poetry.toml").exists());
}

#[test]
fn test_dependency_groups_strategy_include_in_dev() {
    let fixture_path = Path::new(FIXTURES_PATH).join("with_lock_file");

    let tmp_dir = tempdir().unwrap();
    let project_path = tmp_dir.path();

    for file in ["poetry.lock", "poetry.toml", "pyproject.toml"] {
        fs::copy(fixture_path.join(file), project_path.join(file)).unwrap();
    }

    apply_lock_filters!();
    assert_cmd_snapshot!(cli()
        .arg(project_path)
        .arg("--dependency-groups-strategy")
        .arg("include-in-dev"), @r###"
    success: true
    exit_code: 0
    ----- stdout -----

    ----- stderr -----
    Locking dependencies with "uv lock"...
    Using [PYTHON_INTERPRETER]
    Resolved [PACKAGES] packages in [TIME]
    Locking dependencies with "uv lock" again to remove constraints...
    Using [PYTHON_INTERPRETER]
    Resolved [PACKAGES] packages in [TIME]
    Successfully migrated project from Poetry to uv!
    "###);

    insta::assert_snapshot!(fs::read_to_string(project_path.join("pyproject.toml")).unwrap());

    // Assert that previous package manager files are correctly removed.
    assert!(!project_path.join("poetry.lock").exists());
    assert!(!project_path.join("poetry.toml").exists());
}

#[test]
fn test_dependency_groups_strategy_keep_existing() {
    let fixture_path = Path::new(FIXTURES_PATH).join("with_lock_file");

    let tmp_dir = tempdir().unwrap();
    let project_path = tmp_dir.path();

    for file in ["poetry.lock", "poetry.toml", "pyproject.toml"] {
        fs::copy(fixture_path.join(file), project_path.join(file)).unwrap();
    }

    apply_lock_filters!();
    assert_cmd_snapshot!(cli()
        .arg(project_path)
        .arg("--dependency-groups-strategy")
        .arg("keep-existing"), @r###"
    success: true
    exit_code: 0
    ----- stdout -----

    ----- stderr -----
    Locking dependencies with "uv lock"...
    Using [PYTHON_INTERPRETER]
    Resolved [PACKAGES] packages in [TIME]
    Locking dependencies with "uv lock" again to remove constraints...
    Using [PYTHON_INTERPRETER]
    Resolved [PACKAGES] packages in [TIME]
    Successfully migrated project from Poetry to uv!
    "###);

    insta::assert_snapshot!(fs::read_to_string(project_path.join("pyproject.toml")).unwrap());

    // Assert that previous package manager files are correctly removed.
    assert!(!project_path.join("poetry.lock").exists());
    assert!(!project_path.join("poetry.toml").exists());
}

#[test]
fn test_dependency_groups_strategy_merge_into_dev() {
    let fixture_path = Path::new(FIXTURES_PATH).join("with_lock_file");

    let tmp_dir = tempdir().unwrap();
    let project_path = tmp_dir.path();

    for file in ["poetry.lock", "poetry.toml", "pyproject.toml"] {
        fs::copy(fixture_path.join(file), project_path.join(file)).unwrap();
    }

    apply_lock_filters!();
    assert_cmd_snapshot!(cli()
        .arg(project_path)
        .arg("--dependency-groups-strategy")
        .arg("merge-into-dev"), @r###"
    success: true
    exit_code: 0
    ----- stdout -----

    ----- stderr -----
    Locking dependencies with "uv lock"...
    Using [PYTHON_INTERPRETER]
    Resolved [PACKAGES] packages in [TIME]
    Locking dependencies with "uv lock" again to remove constraints...
    Using [PYTHON_INTERPRETER]
    Resolved [PACKAGES] packages in [TIME]
    Successfully migrated project from Poetry to uv!
    "###);

    insta::assert_snapshot!(fs::read_to_string(project_path.join("pyproject.toml")).unwrap());

    // Assert that previous package manager files are correctly removed.
    assert!(!project_path.join("poetry.lock").exists());
    assert!(!project_path.join("poetry.toml").exists());
}

#[test]
fn test_skip_lock() {
    let fixture_path = Path::new(FIXTURES_PATH).join("with_lock_file");

    let tmp_dir = tempdir().unwrap();
    let project_path = tmp_dir.path();

    for file in ["poetry.lock", "poetry.toml", "pyproject.toml"] {
        fs::copy(fixture_path.join(file), project_path.join(file)).unwrap();
    }

    assert_cmd_snapshot!(cli().arg(project_path).arg("--skip-lock"), @r###"
    success: true
    exit_code: 0
    ----- stdout -----

    ----- stderr -----
    Successfully migrated project from Poetry to uv!
    "###);

    insta::assert_snapshot!(fs::read_to_string(project_path.join("pyproject.toml")).unwrap());

    // Assert that previous package manager files are correctly removed.
    assert!(!project_path.join("poetry.lock").exists());
    assert!(!project_path.join("poetry.toml").exists());

    // Assert that `uv.lock` file was not generated.
    assert!(!project_path.join("uv.lock").exists());
}

#[test]
fn test_skip_lock_full() {
    let fixture_path = Path::new(FIXTURES_PATH).join("full");

    let tmp_dir = tempdir().unwrap();
    let project_path = tmp_dir.path();

    fs::copy(
        fixture_path.join("pyproject.toml"),
        project_path.join("pyproject.toml"),
    )
    .unwrap();

    assert_cmd_snapshot!(cli().arg(project_path).arg("--skip-lock"), @r###"
    success: true
    exit_code: 0
    ----- stdout -----

    ----- stderr -----
    warning: Could not find dependency "non-existing-dependency" listed in "extra-with-non-existing-dependencies" extra.
    Successfully migrated project from Poetry to uv!
    "###);

    insta::assert_snapshot!(fs::read_to_string(project_path.join("pyproject.toml")).unwrap());

    // Assert that `uv.lock` file was not generated.
    assert!(!project_path.join("uv.lock").exists());
}

#[test]
fn test_dry_run() {
    let project_path = Path::new(FIXTURES_PATH).join("with_lock_file");
    let pyproject = fs::read_to_string(project_path.join("pyproject.toml")).unwrap();

    assert_cmd_snapshot!(cli().arg(&project_path).arg("--dry-run"));

    // Assert that `pyproject.toml` was not updated.
    assert_eq!(
        pyproject,
        fs::read_to_string(project_path.join("pyproject.toml")).unwrap()
    );

    // Assert that previous package manager files have not been removed.
    assert!(project_path.join("poetry.lock").exists());
    assert!(project_path.join("poetry.toml").exists());

    // Assert that `uv.lock` file was not generated.
    assert!(!project_path.join("uv.lock").exists());
}

#[test]
fn test_dry_run_minimal() {
    let project_path = Path::new(FIXTURES_PATH).join("minimal");
    let pyproject = fs::read_to_string(project_path.join("pyproject.toml")).unwrap();

    assert_cmd_snapshot!(cli().arg(&project_path).arg("--dry-run"));

    // Assert that `pyproject.toml` was not updated.
    assert_eq!(
        pyproject,
        fs::read_to_string(project_path.join("pyproject.toml")).unwrap()
    );

    // Assert that `uv.lock` file was not generated.
    assert!(!project_path.join("uv.lock").exists());
}
