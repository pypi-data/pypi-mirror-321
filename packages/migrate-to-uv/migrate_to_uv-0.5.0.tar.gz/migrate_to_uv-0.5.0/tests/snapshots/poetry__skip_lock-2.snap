---
source: tests/poetry.rs
expression: "fs::read_to_string(project_path.join(\"pyproject.toml\")).unwrap()"
---
[project]
name = "foo"
version = "0.0.1"
requires-python = "~=3.11"
dependencies = ["arrow>=1.2.3,<2"]

[dependency-groups]
dev = ["factory-boy>=3.2.1,<4"]
typing = ["mypy>=1.13.0,<2"]

[tool.uv]
package = false
default-groups = [
    "dev",
    "typing",
]
